packages/omgidl-serialization/src/MessageReader.ts

```diff
@@ -213,6 +218,10 @@ export class MessageReader<T = unknown> {
 
       if (field.typeDeserInfo.type === "struct" || field.typeDeserInfo.type === "union") {
         if (field.isArray === true) {
+          // sequences and arrays have dHeaders only when emHeaders were not already written
+          if (headerOptions.readDelimiterHeader && !readEmHeader) {
+            reader.dHeader();
```

Comment:
why do we ignore the return value?

Reply:
It's not necessary for us since the sequence length is still written. We would use it if we were doing partial deserialization -- in which case we could use it to skip the field entirely